creme/creme_core/static/creme_core/js/widgets/ajax/url.js

```diff
@@ -15,167 +15,32 @@
-    },
-
-    relativeUrl: function() {
-        return '${pathname}${search}${hash}'.template(this._parser);
-    },
-
-    username: function(username) {
-        return this._property('username', username);
-    },
-
-    password: function(password) {
-        return this._property('password', password);
-    },
-
-    protocol: function(protocol) {
-        return this._property('protocol', protocol);
-    },
-
-    host: function(host) {
-        return this._property('host', host);
-    },
-
-    hostname: function(hostname) {
-        return this._property('hostname', hostname);
-    },
-
-    port: function(port) {
-        return this._property('port', port);
-    },
-
-    pathname: function(pathname) {
-        return this._property('pathname', pathname);
-    },
-
-    search: function(search) {
-        if (search === undefined) {
-            return this._parser.search;
-        }
-
-        this._parser.search = search;
-        this._searchData = __decodeSearchData(search);
-        return this;
-    },
-
-    hash: function(hash) {
-        return this._property('hash', hash);
-    },
-
-    properties: function() {
-        return {
-            href: this.href(),
-            username: this.username(),
-            password: this.password(),
-            protocol: this.protocol(),
-            host: this.host(),
-            hostname: this.hostname(),
-            port: this.port(),
-            pathname: this.pathname(),
-            search: this.search(),
-            searchData: this.searchData(),
-            hash: this.hash()
-        };
-    },
-
-    searchData: function(data) {
-        if (data === undefined) {
-            return this._searchData;
-        }
-
-        this.search(creme.ajax.param(data));
-        return this;
-    },
-
-    updateSearchData: function(data) {
-        data = data || {};
-        return this.searchData($.extend({}, this._searchData, data));
-    },
-
-    toString: function() {
-        return this._parser.toString();
-    }
-});
+creme.ajax.URL = RelativeURL;
 
 creme.ajax.parseUrl = function(url) {
-    var parser = document.createElement('a');
-
-    parser.href = url;
-
-    return {
-        href: parser.href,
-        username: parser.username,
-        password: parser.password,
-        protocol: parser.protocol,
-        host: parser.host,
-        hostname: parser.hostname,
-        port: parser.port,
-        pathname: parser.pathname,
-        search: parser.search,
-        searchData: __decodeSearchData(parser.search),
-        hash: parser.hash
-    };
+    console.warn('creme.ajax.parseUrl() is deprecated; Use _.urlAsDict() instead');
+    return _.urlAsDict(url);
 };
 
 creme.ajax.param = function(data) {
     // Use explicit traditional=true argument to replace ajaxSettings.traditional deprecated
     // since jQuery 1.9 see (https://bugs.jquery.com/ticket/12137)
     // return $.param(data, jQuery.ajaxSettings.traditional);
```

Comment:
Comment still relevant?

Reply:
good point ! I can finally get rid of this awful hack :)